components/feature/accounts/src/test/java/mozilla/components/feature/accounts/FxaWebChannelFeatureTest.kt

```diff
@@ -144,17 +150,61 @@ class FxaWebChannelFeatureTest {
         WebExtensionController.installedExtensions[FxaWebChannelFeature.WEB_CHANNEL_EXTENSION_ID] = ext
 
         whenever(sessionManager.getOrCreateEngineSession(session)).thenReturn(engineSession)
-        val webchannelFeature = spy(FxaWebChannelFeature(testContext, null, engine, sessionManager, accountManager))
+        val webchannelFeature = spy(FxaWebChannelFeature(testContext, null, engine, sessionManager, accountManager, serverConfig))
         webchannelFeature.extensionController = controller
 
         webchannelFeature.onSessionAdded(session)
         verify(controller).registerContentMessageHandler(eq(engineSession), messageHandler.capture(), any())
     }
 
+    @Test
+    fun `Ignores messages coming from a different FxA host than configured`() {
+        val sessionManager = mock<SessionManager>()
+        val accountManager = mock<FxaAccountManager>()
+        val serverConfig = mock<ServerConfig>()
+        val session = mock<Session>()
+        val engineSession = mock<EngineSession>()
+        val ext = mock<WebExtension>()
+        val messageHandler = argumentCaptor<MessageHandler>()
+        val port = mock<Port>()
+
+        WebExtensionController.installedExtensions[FxaWebChannelFeature.WEB_CHANNEL_EXTENSION_ID] = ext
+
+        whenever(accountManager.supportedSyncEngines()).thenReturn(null)
+        whenever(sessionManager.getOrCreateEngineSession(session)).thenReturn(engineSession)
+        whenever(port.engineSession).thenReturn(engineSession)
+        whenever(port.senderUrl()).thenReturn("https://bar.foo/email")
+        whenever(serverConfig.contentUrl).thenReturn("https://foo.bar")
```

Comment:
As mentioned above, I think we should test a lot more mis-matched combinations here, but perhaps we can do that in a way that only tests the matching logic rather than having to spin up all this infra and actually send a request.